src/lib/environment/lexical_environment.rs

```diff
@@ -90,6 +99,12 @@ impl LexicalEnvironment {
         self.environment_stack.pop_back();
     }
 
+    pub fn environments(&self) -> impl Iterator<Item = Environment> {
+        std::iter::successors(Some(self.get_current_environment_ref().clone()), |env| {
+            env.borrow().get_outer_environment()
+        })
+    }
+
```

Comment:
nice utility function

Reply:
Thank you, I was quite happy with it :smile: